frontend/src/utilities/js-message-dispatcher.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { reactive } from "vue";
+import { plainToInstance } from "class-transformer";
+import {
+	DisplayConfirmationToCloseAllDocuments,
+	DisplayConfirmationToCloseDocument,
+	DisplayError,
+	DisplayPanic,
+	ExportDocument,
+	newDisplayFolderTreeStructure,
+	OpenDocumentBrowse,
+	SaveDocument,
+	SetActiveDocument,
+	SetActiveTool,
+	SetCanvasRotation,
+	SetCanvasZoom,
+	UpdateCanvas,
+	UpdateOpenDocumentsList,
+	UpdateRulers,
+	UpdateScrollbars,
+	UpdateWorkingColors,
+	UpdateLayer,
+	JsMessage,
+} from "./js-messages";
+
+type JsMessageCallback<T extends JsMessage> = (responseData: T) => void;
+type JsMessageCallbackMap = {
+	[response: string]: JsMessageCallback<any> | undefined;
+};
+
+const state = reactive({
+	responseMap: {} as JsMessageCallbackMap,
+});
+
+type Constructs<T> = new (...args: any[]) => T;
+type ConstructsJsMessage = Constructs<JsMessage> & typeof JsMessage;
+
+const responseMap = {
+	UpdateCanvas,
+	UpdateScrollbars,
+	UpdateRulers,
+	ExportDocument,
+	SaveDocument,
+	OpenDocumentBrowse,
+	DisplayFolderTreeStructure: newDisplayFolderTreeStructure,
+	UpdateLayer,
+	SetActiveTool,
+	SetActiveDocument,
+	UpdateOpenDocumentsList,
+	UpdateWorkingColors,
+	SetCanvasZoom,
+	SetCanvasRotation,
+	DisplayError,
+	DisplayPanic,
+	DisplayConfirmationToCloseDocument,
+	DisplayConfirmationToCloseAllDocuments,
+} as const;
+
+export type JsMessageType = keyof typeof responseMap;
+
+function isResponseConstructor(fn: ConstructsJsMessage | ((data: any) => JsMessage)): fn is ConstructsJsMessage {
+	return (fn as ConstructsJsMessage).responseMarker !== undefined;
+}
+
+export function handleResponse(responseType: JsMessageType, responseData: any) {
+	const MessageMaker = responseMap[responseType];
```

Comment:
Do you think it's correct to use PascalCase here? The variable *contains* a class, but the variable *itself* is not a class. I don't have much of an opinion on the matter, but I thought I'd bring it up to ask yours.

Reply:
I could go either way as well. The more I look at it the more I think it should be camelCase